Functions/Gherkin.Output.ps1

```diff
@@ -0,0 +1,907 @@
+        [PSObject]$StepResult,
+
+        [int]$IndentationLevel
+    )
+
+    Process {
+        $Margin = $Script:ReportStrings.Margin * $IndentationLevel
+
+        $StepResult |
+        Format-StepResultErrorRecord |
+        ForEach-Object { $_.Message -split '\r?\n' } |
+        Select-Object -First 1 |
+        ForEach-Object {
+            "${Margin}$($_ -replace 'Exception:\s*')"
+            $StepResult.StackTrace -replace '(?m)^', $Margin
+        } |
+        & $SafeCommands['Write-Host'] -ForegroundColor $Script:ReportTheme.Pending
+    }
+}
+
+filter Write-UndefinedStepErrorText {
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory = $True, ValueFromPipeline = $True)]
+        [PSObject]$StepResult,
+
+        [int]$IndentationLevel
+    )
+
+    Process {
+        $Margin = $Script:ReportStrings.Margin * $IndentationLevel
+        $WriteHost = $SafeCommands['Write-Host']
+        $Undefined = $Script:ReportTheme.Undefined
+
+        & $WriteHost -ForegroundColor $Undefined $($StepResult.FailureMessage -replace '(?m)^', "${Margin}  >>> ")
+        & $WriteHost -ForegroundColor $Undefined $($StepResult.StackTrace -replace '(?m)^', "$Margin  ")
+    }
+}
+
+function Format-StepResultErrorRecord {
+    [CmdletBinding()]
+    [OutputType([string])]
+    Param (
+        [Parameter(Position = 0, Mandatory = $True, ValueFromPipeline = $True)]
+        [PSObject]$StepResult
+    )
+
+    Process {
+        $ErrorRecord = $StepResult.ErrorRecord
+
+        $ErrorLines = [PSCustomObject]@{
+            Message = @()
+            Trace   = @()
+        }
+
+        ## convert the exception messages
+        $Exception = $ErrorRecord.Exception
+        $ExceptionLines = @()
+
+        while ($Exception) {
+            $ExceptionName = $Exception.GetType().Name
+            $MessageLines = $Exception.Message.Split([string[]]($([System.Environment]::NewLine), "\n", "`n"), [System.StringSplitOptions]::RemoveEmptyEntries)
+
+            if ($ErrorRecord.FullyQualifiedErrorId -ne 'PesterAssertionFailed') {
+                $MessageLines[0] = "${ExceptionName}: $($MessageLines[0])"
+            }
+
+            [array]::Reverse($MessageLines)
+            $ExceptionLines += $MessageLines
+            $Exception = $Exception.InnerException
+        }
+
+        [array]::Reverse($ExceptionLines)
+
+        $ErrorLines.Message += $ExceptionLines
+
+        if ('PesterAssertionFailed', 'PesterGherkinStepFailed' -contains $ErrorRecord.FullyQualifiedErrorId) {
+            $ErrorLines.Message += "$($ErrorRecord.TargetObject.Line)`: $($ErrorRecord.TargetObject.LineText.Trim())".Split([string[]]($([System.Environment]::NewLine), '\n', "`n"), [System.StringSplitOptions]::RemoveEmptyEntries)
+            $ErrorLines.Trace += @($StepResult.StackTrace -split '\r?\n')
+        }
+
+        $ErrorLines
+
+        # ! Leaving this here because I may need this if it becomes evident that showing the portion of the
```

Comment:
I'm interested in knowing if I need to put any of this code back in in any way? Near as far as I can tell, the rest of the stack trace usually contained Pester internals information, which isn't really helpful to someone in trying to understand why their test is failing. (Though, I must admit, the reason why I started working on Pester is because Pester internals (and my lack of PS knowledge) were causing my tests to fail! Lol...ironic.)

Reply:
OK, I'm going to reinstate these commented-out lines. But I have a feeling that the stack trace is not going to show any further details due to how I'm processing the stack trace in `ConvertTo-GherkinStepResult'. I'll need to rediscover 1) why I commented out those lines to begin with (it had something to do with getting consistent output for use with VS Code Task problem matchers without them breaking every time someone changed something in this area of the codebase); and 2) what the behavior ought to be when the code breaks in the code/system under test, in a script block and in a feature. (The last two are already working fine, but what does/should the stack trace look like for broken code (e.g. NullReferenceException, InvalidOperationException, etc.)?)

Reply:
Alright. I think I found the error of my ways. (Pun intended.) I'll need to do a bit of legwork on this section here, but given this PR is fairly large, it's still good to get this review going, and this is a relatively small section (but nonetheless important).

Reply:
OK, I have been able to reinstate these commented-out lines. However, their location has changed. And in the process, after reorganizing the code, I've been able to remove ~400 lines of code. This will be resolved with the next commit. So I'm going to go ahead and resolve this now.